blacklistfilter/evaluator/evaluator.py

```diff
@@ -0,0 +1,316 @@
+import csv
+import json
+from queue import Queue
+import os
+import sys
+import signal
+import time
+from threading import Thread, Event
+import requests
+from datetime import datetime
+import pytrap
+import shutil
+from optparse import OptionParser
+from collections import defaultdict
+
+#Enter an API key to obtain informations from nerd
+NERD_API_KEY = None
+
+#Constant defined in Adaptive Filter
+ADAPTIVE_BLACKLIST_ID = 999
+
+parser = OptionParser(add_help_option=True)
+parser.add_option("-i", "--ifcspec", dest="ifcspec",
+                  help="TRAP IFC specifier", metavar="IFCSPEC")
+parser.add_option("-c", "--csv-path", dest="csv", help="Path to the csv files generated by Split Evidence")
+parser.add_option("-e","--evidence-path", dest="evidence", help="Path to the folder for storing data of unreported detections")
+
+#Signal handler
+def SignalHandler(signal, frame):
+    global signal_recieved, trap_ctx
+    if signal_recieved:
+        print('Caught another SIGINT, exiting forcefully..')
+        os._exit(1)
+    trap_ctx.terminate()
+    signal_recieved = True
+
+class Alert:
+    def __init__(self, client, nerd_info):
+        self.ip = client.ip_addr;
+        self.statistics = client.statistics
+        self.nerd_info = nerd_info
+
+class MonitoredClient:
+
+    def __init__(self, ip_addr, detection):
+        self.ip_addr = ip_addr
+        self.detection = detection
+        self.ports = set()
+        self.duration_in = 0.0
+        self.duration_out = 0.0
+        self.start = datetime.now()
+        self.end = datetime.utcfromtimestamp(0)
+        self.ips_contacted = set()
+        self.statistics = defaultdict(int)
+
+    def AddFlow(self, flow, src):
+        first = datetime.strptime(flow["time TIME_FIRST"], '%Y-%m-%dT%H:%M:%S.%f')
+        last = datetime.strptime(flow["time TIME_LAST"], '%Y-%m-%dT%H:%M:%S.%f')
+
+        if first < self.start:
+            self.start = first
+        if last > self.end:
+            self.end = last
+
+        if src:
+            self.duration_out += (last - first).total_seconds();
+            self.statistics["bytes_sent"] += int(flow["uint64 BYTES"])
+            self.statistics["packets_sent"] += int(flow["uint32 PACKETS"])
+            self.statistics["flows_sent"] += 1
+            self.ips_contacted.add(flow["ipaddr DST_IP"])
+            self.ports.add(flow["uint16 DST_PORT"])
+        else:
+            self.duration_in += (last - first).total_seconds();
+            self.statistics["bytes_recv"] += int(flow["uint64 BYTES"])
+            self.statistics["packets_recv"] += int(flow["uint32 PACKETS"])
+            self.statistics["flows_recv"] += 1
+
+    #Determine other statistics
```

Comment:
What does it mean `Determine other statistics`? Do you mean "compute statistics for ... client? + Doc should be written according to https://www.python.org/dev/peps/pep-0257/